### PR TITLE
chore(Core-Frontend): Run create-react-class codemod

### PR DESCRIPTION
**Pull request by MaxBittker**
In cases where we can't use es6 class syntax because of mixins, this PR converts us to use the standalone shim "create-react-class"

This needs to be followed up with a getsentry codemod 

helps us get closer to react 16:
https://github.com/getsentry/sentry/issues/6495

### CODE REVIEW

File: src/sentry/static/sentry/app/components/actionOverlay.jsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
...
 import React from 'react';
+import createReactClass from 'create-react-class';
 import OrganizationState from '../mixins/organizationState';
 import {t} from '../locale';
 import requiredAdminActions from '../components/requiredAdminActions';
 import LoadingIndicator from '../components/loadingIndicator';
 
-const ActionOverlay = React.createClass({
+const ActionOverlay = createReactClass({
+  displayName: 'ActionOverlay',
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/actionOverlay.jsx]: 🤮

File: src/sentry/static/sentry/app/views/groupDetails/actions.jsx [hunk 2]
```diff
@@ -18,26 +20,24 @@ import TooltipMixin from '../../mixins/tooltip';
...
     onDiscard: PropTypes.func.isRequired,
-  },
+  };
 
-  getInitialState() {
-    return {
-      hooksDisabled: HookStore.get('project:discard-groups:disabled'),
-    };
-  },
+  state = {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/groupDetails/actions.jsx]: Let's make sure we initialize state in constructor to be consistent

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/groupDetails/actions.jsx (hunk 2 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/groupDetails/actions.jsx]: It's on the ~4 components that are now es6 components. Unfortunately I think we have to do it manually.

File: src/sentry/static/sentry/app/components/actionOverlay.jsx (hunk 1 above)
Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/actionOverlay.jsx]: 😬

File: src/sentry/static/sentry/app/views/organizationStats/index.jsx [hunk 3]
```diff
@@ -146,11 +153,13 @@ const OrganizationStats = createReactClass({
...
-          this.state.projectsRequestsPending -= 1;
-          this.setState({
-            rawProjectData: this.state.rawProjectData,
-            projectsRequestsPending: this.state.projectsRequestsPending,
+          let rawProjectData = this.state;
+          rawProjectData[statName] = data;
+          this.setState(state => {
+            return {
+              rawProjectData,
+              projectsRequestsPending: this.state.projectsRequestsPending - 1,
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationStats/index.jsx]: `this.state` --> `state`

File: src/sentry/static/sentry/app/views/organizationStats/index.jsx [hunk 4]
```diff
@@ -122,11 +126,14 @@ const OrganizationStats = createReactClass({
           stat: statName,
         },
         success: data => {
-          this.state.rawOrgData[statName] = data;
-          this.state.statsRequestsPending -= 1;
-          this.setState({
-            rawOrgData: this.state.rawOrgData,
-            statsRequestsPending: this.state.statsRequestsPending,
+          this.setState(state => {
+            let rawOrgData = this.state;
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/organizationStats/index.jsx]: s/this.//

File: src/sentry/static/sentry/app/views/organizationStats/index.jsx (hunk 3 above)
Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/organizationStats/index.jsx]: this is a good reason to use "prevState" as the name. fixed a few bugs in this code thanks to the integration test

Comment (billyvg) [reply src/sentry/static/sentry/app/views/organizationStats/index.jsx]: it's not really previous state though, it's more current state

Comment (MaxBittker) [reply src/sentry/static/sentry/app/views/organizationStats/index.jsx]: ... true ... but it's close enough right? willing to change it if you have an idea

**Comment by MaxBittker** [comment]
fixed the test, some of the only code i manually wrote in this PR and it had like 4 bugs 😭

**Review by billyvg — APPROVED** [review]
We'll want to:
* export `createReactClass` for getsentry
* tracking PR on gs
* codemod on gs
* test on staging

**Review by lynnagara — APPROVED** [review]

**Pull request merged**